src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,271 @@
+package duke;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Represents the Duke chatbot.
+ */
+public class Duke {
+
+    private static final String FILEPATH = "./data/duke.txt";
+    private static boolean hasNext = true;
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
```

Comment:
Good job on using the appropriate naming convention for variables throughout your code :+1: